### PR TITLE
Edit description of topics

### PR DESCRIPTION
**Pull request by PolatEmre-hg**
This Pull Request implements the feature of changing the Description of a given Topic after its creation. You can only change the  Description if you own the Topic and you are not on the staged Environment, that is if you have staged a Topic from Dev to Prod you can only change the Description on Dev, on the prod Environment the Button for editing the Description will not appear!

### CODE REVIEW

File: src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java [hunk 1]
```diff
@@ -780,22 +784,38 @@ public void testChangeDescOfTopic() throws Exception {
...
 	public void testDontResetDeprecationWhenTopicDescChanges() throws Exception {
 
+		SubscriptionService subscriptionService = mock(SubscriptionService.class);
+
+		TopicServiceImpl service = new TopicServiceImpl(kafkaClusters, applicationsService, topicNameValidator,
+				userService, topicConfig, eventManager);
+		ValidatingTopicServiceImpl validatingService = new ValidatingTopicServiceImpl(service, subscriptionService,
+				applicationsService, kafkaClusters, topicConfig);
+
+		UpdateTopicDto dto = new UpdateTopicDto("topic is now deprecated",LocalDate.of(2199,2,14),
```
Comment (albrechtflo-hg) [inline src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java:794]: Missing space before comma (check formatter settings)

File: src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java [hunk 2]
```diff
@@ -780,22 +784,38 @@ public void testChangeDescOfTopic() throws Exception {
...
+				"updated description goes here", true);
+
 		TopicMetadata topic = new TopicMetadata();
 		topic.setName("topic-1");
+		topic.setDeprecated(true);
+		topic.setEolDate(LocalDate.of(2299,12,4));
```
Comment (albrechtflo-hg) [inline src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java:800]: Missing spaces (check formatter)

**Review by albrechtflo-hg — CHANGES_REQUESTED** [review]

File: src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java (hunk 1 above)
Comment (PolatEmre-hg) [reply src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java]: Code is now formatted.

File: src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java (hunk 2 above)
Comment (PolatEmre-hg) [reply src/test/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImplTest.java]: Code is now formatted.

File: ui/src/app/layout/topics/single-topic.component.html [hunk 3]
```diff
@@ -27,6 +27,14 @@ <h2 class="text-muted d-inline-block">{{ 'Topic' | translate }} {{ topicName | a
                                        <span class="deprecatedNote">Deprecated</span>
                                         <span class="deprecationText">{{(topic | async)?.deprecationText}}</span>
                                         </span>
+                                        <button
+                                            *ngIf="(isOwnerOfTopic | async) && !(selectedEnvironment | async)?.stagingOnly"
+                                            appSpinnerWhile
+                                            #p="appSpinnerWhile"
+                                            (click)="(openDeleteConfirmDlg(updateTopicDescription))"
```
Comment (albrechtflo-hg) [inline ui/src/app/layout/topics/single-topic.component.html:34]: If you re-use `openDeleteConfirmDlg`for opening the dialog, you should rename that method ;-) Oh, and the outer braces can be removed.

File: ui/src/app/layout/topics/single-topic.component.html [hunk 4]
```diff
@@ -450,3 +458,24 @@ <h4 class="modal-title">{{ 'Delete Schema' | translate }}</h4>
...
+<ng-template #updateTopicDescription let-dlg>
+    <div class="modal-header">
+        <h4 class="modal-title">{{ 'Edit' | translate }}</h4>
+        <button type="button" class="close" aria-label="Close" (click)="dlg.dismiss()">
+            <span aria-hidden="true">&times;</span>
+        </button>
+    </div>
+    <div class="modal-body">
+        <div class="mt-3">
+            <textarea class="form-control" rows="5" cols="11" [(ngModel)]="updatedTopicDescription"></textarea></div>
```
Comment (albrechtflo-hg) [inline ui/src/app/layout/topics/single-topic.component.html:471]: Where is `updatedTopicDescription` initialized? Looks to me as if the textarea is empty, even if the topic already had a description before... (only from looking at the code, so maybe I am missing something)

File: ui/src/app/layout/topics/single-topic.component.html (hunk 3 above)
Comment (PolatEmre-hg) [reply ui/src/app/layout/topics/single-topic.component.html]: done.

File: ui/src/app/layout/topics/single-topic.component.html (hunk 4 above)
Comment (PolatEmre-hg) [reply ui/src/app/layout/topics/single-topic.component.html:471]: done.

**Review by albrechtflo-hg — APPROVED** [review]
LGTM

**Pull request merged**